### PR TITLE
properly determine go versions

### PR DESCRIPTION
**Pull request by deitch**
This extracts the unique `pkg/pillar/scripts/getversion.sh` and replaces it with `tools/goversion.sh`, which calculates the actual version for a go package using the go semver + pseudo-version rules. Unfortunately, no standard go tool does this for you.

`getversion.sh` was being used only to set `-X main.Version=<version>` for pillar. That version was interesting, and didn't actually confirm to go rules.

This now uses the standard version at a higher level, and then includes it in the 3 go binaries we build: edge-view, newlog, pillar.

The process is as follows:

1. In `Makefile`, have every `eve-<pkg>` target depend upon `pkg/%/gopkgversion`
2. In `Makefile`, the `pkg/%/gopkgversion` generates the version from `goversion.sh` and saves it in the package in `pkg/%/gopkgversion`
3. `*gopkgversion` has been added to `.gitignore`
4. In the Dockerfiles and/or Makefiles that build binaries, the ldflag `-X main.Version=<version>` has been updated to source from that file.

The `gopkgversion` non-tracked file allows us to build in subdirs like `pkg/pillar/` or `pkg/newlog/` without requiring the entire `.git` directory in the context.

### CODE REVIEW

File: pkg/newlog/Dockerfile [hunk 1]
```diff
@@ -6,7 +6,7 @@ RUN eve-alpine-deploy.sh
 COPY ./  /newlog/.
 WORKDIR /newlog
 
-RUN GO111MODULE=on CGO_ENABLED=0 go build -ldflags "-s -w" -mod=vendor -o /out/usr/bin/newlogd ./cmd
+RUN GO111MODULE=on CGO_ENABLED=0 go build -ldflags "-s -w -X=main.Version=$(cat gopkgversion)" -mod=vendor -o /out/usr/bin/newlogd ./cmd
```
Comment (christoph-zededa) [inline pkg/newlog/Dockerfile:9]:
```suggestion
RUN GO111MODULE=on CGO_ENABLED=0 go build -ldflags "-s -w -X=main.Version=$(<gopkgversion)" -mod=vendor -o /out/usr/bin/newlogd ./cmd
```

Comment (deitch) [reply pkg/newlog/Dockerfile:9]: It is a bashism. I originally had that, and spent easily 20-30 mins trying to figure out why the ldflag was blank. If I had submitted it right away, yetus would have told me, but I insisted on figuring it out myself first. 😆

File: tools/goversion.sh [hunk 2]
```diff
@@ -0,0 +1,24 @@
+#!/bin/sh
+#
+# Copyright (c) 2018 Zededa, Inc.
```
[the remark this answers is not in the document]
Comment (deitch) [reply tools/goversion.sh:3]: LOL! I copied over the `getversion.sh` and then modified it. Clearly, not quite enough. 😆 

I will fix it.

[the remark this answers is not in the document]
Comment (deitch) [reply tools/goversion.sh]: Done

**Review by eriknordmark — APPROVED** [review]
LGTM

I wonder if we take this opportunity to do the eve-version in similar ways e.g., by having the Makefile drop the ROOTFS_VERSION into ./eve-version and then use that. But this can be a separate PR.

**Comment by deitch** [comment]
I am happy to get a PR in for that.

We also were thinking of doing that for kernel builds, so we can avoid some of the complex templating we do. I will try top get to a PR on that too.

**Review by milan-zededa — APPROVED** [review]

**Pull request merged**